lib/pacemaker/pcmk_cluster_queries.c

```diff
@@ -409,28 +434,60 @@ pcmk__list_nodes(pcmk__output_t *out, gboolean BASH_EXPORT)
     rc = the_cib->cmds->query(the_cib, NULL, &xml_node,
                               cib_scope_local | cib_sync_call);
     if (rc == pcmk_ok) {
-        int found = 0;
         xmlNode *node = NULL;
         xmlNode *nodes = get_object_root(XML_CIB_TAG_NODES, xml_node);
+        struct node_data data = {
+            .out = out,
+            .found = 0,
+            .BASH_EXPORT = BASH_EXPORT
+        };
 
         out->begin_list(out, NULL, NULL, "nodes");
 
-        for (node = first_named_child(nodes, XML_CIB_TAG_NODE); node != NULL;
-             node = crm_next_same_xml(node)) {
-            const char *node_type = BASH_EXPORT ? NULL :
-                         crm_element_value(node, XML_ATTR_TYPE);
-            out->message(out, "crmadmin-node", node_type,
-                         crm_str(crm_element_value(node, XML_ATTR_UNAME)),
-                         crm_str(crm_element_value(node, XML_ATTR_ID)),
-                         BASH_EXPORT);
+        if (!pcmk__str_empty(node_types) && strstr(node_types, "all")) {
+            node_types = NULL;
+        }
+
+        if (pcmk__str_empty(node_types) || strstr(node_types, "member")) {
+            for (node = first_named_child(nodes, XML_CIB_TAG_NODE); node != NULL;
+                 node = crm_next_same_xml(node)) {
+                const char *node_type = crm_element_value(node, XML_ATTR_TYPE);
+                //if (node_type == NULL || !strcmp(node_type, "member")) {
+                if (node_type == NULL) {
```

Comment:
Pacemaker Remote nodes can be in this section if they have permanent node attributes